src/Sarif/Converters/FileInfoFactory.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis.Sarif.Converters
+{
+    internal class FileInfoFactory
+    {
+        private readonly Dictionary<string, FileReference[]> _fileInfoDictionary;
+        private readonly Func<string, string> _mimeTypeDeterminer;
+
+        internal FileInfoFactory(Func<string, string> mimeTypeDeterminer)
+        {
+            _mimeTypeDeterminer = mimeTypeDeterminer;
```

Comment:
'determiner'. sounds suspiciously like a new coinage (for those of us who aren't grammer experts, where this is a thing). :) how about 'mimeTypeClassifier' ?